### PR TITLE
Resolve NEXT_MAJOR comments

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for next major.

## Changelog

```markdown
### Added
- Added `EditableRolesBuilderInterface`.

### Changed
- Changed `__construct()` of `EditableRolesBuilder` to provide `TranslatorInterface`
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Tests need to be updated
- Some private methods were tested
- Some mock were made on final classes

**Comment by VincentLanglet** [comment]
For the label, I think the `major` label require to have a Changelog.
So as a hack, we have to use the pedantic label in order to generate the release changelog with the dev kit command.

**Comment by jordisala1991** [comment]
> For the label, I think the `major` label require to have a Changelog. So as a hack, we have to use the pedantic label in order to generate the release changelog with the dev kit command.

I have added the changelog, had to introduce a new interface.

I didn't touch a metod that should be private because I will be fixing that on another PR. (I noted it on the 5.0 roadmap issue)

File: src/Security/EditableRolesBuilder.php [hunk 1]
```diff
@@ -42,23 +39,22 @@ class EditableRolesBuilder
...
      * @param array<string, array<string>> $rolesHierarchy
      */
-    public function __construct(TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker, Pool $pool, SonataConfiguration $configuration, array $rolesHierarchy = [])
-    {
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        AuthorizationCheckerInterface $authorizationChecker,
+        Pool $pool,
+        SonataConfiguration $configuration,
+        ?TranslatorInterface $translator = null,
```
Comment (VincentLanglet) [inline src/Security/EditableRolesBuilder.php]: Translator can be removed isnt it ? It seems not used

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Security/EditableRolesBuilder.php (hunk 1 above)
Comment (jordisala1991) [reply src/Security/EditableRolesBuilder.php]: Is it used, check below.

Comment (VincentLanglet) [reply src/Security/EditableRolesBuilder.php]: Indeed, on the last lines.

Test need to be updated and this should be ok.

Comment (jordisala1991) [reply src/Security/EditableRolesBuilder.php]: Now, they should be okay.

**Review by VincentLanglet — DISMISSED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**